src/pipx/constants.py

```diff
@@ -30,6 +30,48 @@ def is_windows() -> bool:
 WINDOWS: bool = is_windows()
 
 
+# If we are running under the Windows embeddable distribution,
+# venv isn't available (and we probably don't want to use the
+# embeddable distribution as our applications' base Python anyway)
+# so we try to locate the system Python and use that instead.
+
+# This code was copied from https://github.com/uranusjr/pipx-standalone
+# which uses this technique to build a completely standalone pipx
+# distribution
+
+
+def _find_default_python():
+    import shutil
+
+    py = shutil.which("py")
+    if py:
+        return py
+    python = shutil.which("python")
+    if "WindowsApps" not in python:
+        return python
+    # Special treatment to detect Windows Store stub.
+    # https://twitter.com/zooba/status/1212454929379581952
+    import subprocess
+
+    proc = subprocess.run(
+        [python, "-V"], stdout=subprocess.PIPE, stderr=subprocess.DEVNULL,
+    )
+    if proc.returncode != 0:
+        # Cover the 9009 return code pre-emptively.
+        raise EnvironmentError("no available Python found")
```

Comment:
Though `EnvironmentError` is technically accurate, pipx only catches `PipxError`s (at the bottom of `main.py`) and prints a single line, user-facing error. This will result in a full stack trace, which we want to avoid. Could these be changed to `PipxError`s?

Reply:
Also done

Reply:
Hmm, this failed because of a circular import - constants imports util which in turn imports constants. I'll need to look in more detail at how to address this. I'll come back with a revised patch in a while...

Reply:
Actually, after looking for a while for ways to avoid the circular import it occurred to me that there's no point. If we can't find a suitable value for `DEFAULT_PYTHON`, that will happen when constants.py is *imported*, which is outside the scope of the try..except that catches PipxError.

I see two possible fixes here:

1. (Simple, but a bit of a hack) Replace the exception with a simple `raise SystemExit("Python interpreter does not support venv")`. This will avoid an unfriendly traceback, but will remain outside of the general `PipxError` handling.
2. (More complex, maybe better?) Replace DEFAULT_PYTHON with a function, called when needed. I'd propose splitting this out into a separate file, `pipx.interpreter`, to break the circular import.

With option (2) it would also be possible to add some simple unit tests (monkeypatch `import venv` to fail, and check the default python is not `sys.executable`). But we'd be adding some non-trivial complexity for a fairly niche case.

What would you prefer?

Reply:
@cs01 Gentle ping - which approach do you thing I should take with this?

Reply:
Hi sorry for the slow response. 

I lean toward option 2. This will probably not be the last time we need to add additional logic to determining DEFAULT_PYTHON.

Reply:
No worries, I'll see if I can implement that approach. I just didn't want to over-engineer for what is a relatively rare case, without knowing whether you supported it in principle.